pyciemss/ouu/ouu.py

```diff
@@ -107,15 +107,17 @@ def __call__(self, x):
                 "Selected interventions are out of bounds. Will use a penalty instead of estimating risk."
             )
             risk_estimate = max(
-                2 * self.risk_bound, 10.0
+                np.max(2 * np.array(self.risk_bound)), 10.0
             )  # used as a penalty and the model is not run
         else:
             # Apply intervention and perform forward uncertainty propagation
             samples = self.propagate_uncertainty(x)
             # Compute quanity of interest
-            sample_qoi = self.qoi(samples)
+            sample_qoi = [q(samples) for q in self.qoi]
             # Estimate risk
-            risk_estimate = self.risk_measure(sample_qoi)
+            risk_estimate = np.full(len(self.qoi), np.nan)
+            for i in range(len(self.qoi)):
+                risk_estimate[i] = self.risk_measure[i](sample_qoi[i])
```

Comment:
@anirban-chaudhuri , how is this now expanded risk_estimate array used downstream? I'm having trouble tracking how this propagates to the optimizer, and what the implication is for these multiple risk measures?

My understanding based on our many conversations and earlier reviews is that these risk measures are only used for constraints, and that the scipy optimizer accepts arrays as constraints, checking if **any** element is violated. Is that correct?

Reply:
Yes, the risk_estimate is an array now and scipy can handle the numpy array to create multiple constraints.